include/vinecopulib/misc/tools_stats.hpp

```diff
@@ -52,6 +54,52 @@ qnorm(const Eigen::MatrixXd& x)
   return tools_eigen::unaryExpr_or_nan(x, f);
 }
 
+//! @brief Quantile function of the Standard normal distribution with additional
+//! bound checks for numerical stability.
+//!
+//! @param x Evaluation points.
+//!
+//! @return An \f$ n \times d \f$ matrix of evaluated quantiles.
+inline Eigen::MatrixXd
+safe_qnorm(const Eigen::MatrixXd& x)
+{
+  boost::math::normal dist;
+  auto f = [&dist](double y) {
+    if (y <= 0) {
+      return -std::numeric_limits<double>::infinity();
+    } else if (y >= 1) {
+      return std::numeric_limits<double>::infinity();
+    } else {
+      return boost::math::quantile(dist, y);
+    }
+  };
+
+  return tools_eigen::unaryExpr_or_nan(x, f);
+}
+
+//! @brief Distribution function of the Standard normal distribution with
+//! additional bound checks for numerical stability.
+//!
+//! @param x Evaluation points.
+//!
+//! @return An \f$ n \times d \f$ matrix of evaluated quantiles.
+inline Eigen::MatrixXd
+safe_pnorm(const Eigen::MatrixXd& x)
```

Comment:
* Same question as above regarding the bounds.
* Shouldn't you use the formula along with std::erf ?

Reply:
it's actually even better. our new qnorm/pnorm variants already do proper handling of the edge cases, so the safe variants can be removed.